app/code/core/Mage/Core/Model/Layout.php

```diff
@@ -170,7 +170,7 @@ public function generateXml()
                             continue;
                         }
                         if (!isset($block->attributes()->ignore)) {
-                            $block->addAttribute('ignore', true);
+                            $block->addAttribute('ignore', 'true');
```

Comment:
Ref https://www.php.net/manual/en/language.types.declarations.php#:~:text=Strict%20typing%20¶,on%20a%20per-file%20basis:
> By default, PHP will coerce values of the wrong type into the expected scalar type declaration if possible. For example, a function that is given an int for a parameter that expects a string will get a variable of type string.

So `(string) true` is "1".
```suggestion
                            $block->addAttribute('ignore', '1');
```

Reply:
Why not.